authorization/v2/uniformResourceIdentifier.go

```diff
@@ -24,6 +24,64 @@ func NewUriFromRequest(resource resources.Resource, handler gin.HandlerFunc, ctx
 	}
 }
 
+// NewURIFromString parses the string representation of a URI into the UniformResourceIdentifier struct.
+// NewURIFromString expects the string to be of the following form, otherwise ErrInvalidURI is returned.
+// hs:<service_name>:<subsystem>:<version>:<category>:<resource_name>?<allowed_arguments>#<permission_metadata>
+func NewURIFromString(source string) (UniformResourceIdentifier, error) {
+	var (
+		arguments map[string]string
+		metadata  map[string]string
+	)
+
+	pathArgumentSplit := strings.Split(source, "?")
+	path := pathArgumentSplit[0]
+
+	if len(pathArgumentSplit) > 2 {
+		return UniformResourceIdentifier{}, ErrInvalidURI
+	}
+
+	if len(pathArgumentSplit) == 2 {
+		var err error
+
+		argumentsMetadataSplit := strings.Split(pathArgumentSplit[1], "#")
+		arguments, err = unmarshalArguments(argumentsMetadataSplit[0])
```

Comment:
If you changed this to `arguments, err := unmarshalArguments(argumentsMetadataSplit[0])` the `var err error` above would not be required

Reply:
The `var err error` is still required since I define both the `arguments` and `metadata` maps at the start of the function

Reply:
Right, yeah, sorry